### PR TITLE
enable `size` middleware for popovers

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

This PR also stems from #2187 and handles an edge case (no pun intended) of #2155.

**TL;DR**: I've enabled the floating-ui [`size`](https://floating-ui.com/docs/size) middleware by default and used it to set `max-block-size` based on the available height.

This is necessary because we now use `position: fixed`, which positions the popovers based on viewport but can still cause the popovers to overflow the viewport (which is especially problematic in cases where the viewport isn't scrollable). To guard against this, the popovers now stay contained within the viewport.

One of the interesting decisions I had to make is regarding the max-height. I can't just set it to `availableHeight`, or it would become way too tall on larger viewports. So I use `400px` as the default (for `Popover`) and override it to `var(--iui-menu-max-height)` in components that use `List`.

## Testing

Manually verified that the element stays constrained by the viewport on small viewports and high zoom levels.

<details><summary>Screen recordings</summary>

| Before | After |
| --- | --- |
| <video src="https://github.com/user-attachments/assets/3aec1cea-a3c4-4923-9f7a-2507c3893eab"> | <video src="https://github.com/user-attachments/assets/e638b299-00d7-4721-97ac-fb697e453d00"> |
| <video src="https://github.com/user-attachments/assets/4d602d1e-efbb-4b0a-b4c5-e6e908a3eafb"> | <video src="https://github.com/user-attachments/assets/9438193a-5264-4b5e-bbea-247257efff2e"> |

</details> 

## Docs

Added `minor` changeset.

Updated existing popover docs to mention that `size` middleware is enabled by default.

### CODE REVIEW

File: .changeset/young-mayflies-listen.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@itwin/itwinui-react': minor
+---
+
+`Popover` now enables the [`size` middleware](https://floating-ui.com/docs/size) to prevent it from overflowing the viewport.
```
[the remark this answers is not in the document]
Comment (mayank99) [reply .changeset/young-mayflies-listen.md:5]: See next line where I wrote "This also affects other popover-like components (e.g. `Select`, `ComboBox`, `DropdownMenu`)"

**Review by smmr-dn — APPROVED** [review]
